File: app/src/main/java/rk/information/news/network/APIImpl.java
package rk.information.news.network;


import rk.information.news.network.responseModel.AllNewsRes;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by Ritesh on 04/09/2019.
 */

public interface APIImpl {

    // Get all news (10 news items at a time , source = "bbc-news")
    @GET("/v2/everything?sources=bbc-news&pageSize=10")
    Call<AllNewsRes> getAllNews(@Query("page") int pageNum);


}

File: app/src/main/java/rk/information/news/network/APIImpl.kt
package rk.information.news.network

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query
import rk.information.news.network.responseModel.AllNewsRes

/**
 * Created by Ritesh on 04/09/2019.
 */
interface APIImpl {
    // Get all news (10 news items at a time , source = "bbc-news")
    @GET("/v2/everything?pageSize=20&page=1&sortBy=publishedAt")
    fun getAllNews(
            @Query("q") searchQuery: String?,
            @Query("from") from: String?): Call<AllNewsRes?>?
}